contracts/ForwarderV4.sol

```diff
@@ -0,0 +1,380 @@
+   */
+  function init(
+    address _parentAddress,
+    address _feeAddress,
+    bool _autoFlush721,
+    bool _autoFlush1155
+  ) external onlyUninitialized {
+    require(_parentAddress != address(0x0), 'Invalid parent address');
+    parentAddress = _parentAddress;
+    require(_feeAddress != address(0x0), 'Invalid fee address');
+    feeAddress = _feeAddress;
+
+    uint256 value = address(this).balance;
+
+    /// @notice set whether we want to automatically flush erc721/erc1155 tokens or not
+    autoFlush721 = _autoFlush721;
+    autoFlush1155 = _autoFlush1155;
+
+    if (value == 0) {
+      return;
+    }
+
+    (bool success, ) = parentAddress.call{ value: value }('');
+    require(success, 'Flush failed');
+
+    /**
+     * Since we are forwarding on initialization,
+     * we don't have the context of the original sender.
+     * We still emit an event about the forwarding but set
+     * the sender to the forwarder itself
+     */
+    emit ForwarderDeposited(address(this), value, msg.data);
+  }
+
+  /**
+   * @inheritdoc IForwarderV4
+   */
+  function setAutoFlush721(bool autoFlush)
+    external
+    virtual
+    override
+    onlyAllowedAddress
+  {
+    autoFlush721 = autoFlush;
+  }
+
+  /**
+   * @inheritdoc IForwarderV4
+   */
+  function setAutoFlush1155(bool autoFlush)
+    external
+    virtual
+    override
+    onlyAllowedAddress
+  {
+    autoFlush1155 = autoFlush;
+  }
+
+  /**
+   * ERC721 standard callback function for when a ERC721 is transfered. The forwarder will send the nft
+   * to the base wallet once the nft contract invokes this method after transfering the nft.
+   *
+   * @param _operator The address which called `safeTransferFrom` function
+   * @param _from The address of the sender
+   * @param _tokenId The token id of the nft
+   * @param data Additional data with no specified format, sent in call to `_to`
+   */
+  function onERC721Received(
+    address _operator,
+    address _from,
+    uint256 _tokenId,
+    bytes memory data
+  ) external virtual override returns (bytes4) {
+    if (autoFlush721) {
+      IERC721 instance = IERC721(msg.sender);
+      require(
+        instance.supportsInterface(type(IERC721).interfaceId),
+        'The caller does not support the ERC721 interface'
+      );
+      /// this won't work for ERC721 re-entrancy
+      instance.safeTransferFrom(address(this), parentAddress, _tokenId, data);
+    }
+
+    return this.onERC721Received.selector;
+  }
+
+  /**
+   * @notice Method to allow for calls to other contracts. This method can only be called by the allowed address
+   * @param target The target contract address whose method needs to be called
+   * @param value The amount of Ether to be sent
+   * @param data The calldata to be sent
+   */
+  function callFromAllowedAddress(
+    address target,
+    uint256 value,
+    bytes calldata data
+  ) external onlyAllowedAddress returns (bytes memory) {
```

Comment:
wondering do we have to allow `feeAddress` here as well ?
cause using feeAddr we can move funds to any address

Reply:
I think that's a valid point. Bitgo on its own shouldn't be able to move any funds. This function should only be used in exceptional cases any ways

Reply:
updated this, now only parent address will be able to call this method